Test flake in CI on TestReactorValidatorSetChanges

## Repro

I have not yet had success reproducing this locally, but this is the command to run the test that flakes in CI.

```sh
go test -run TestReactorValidatorSetChanges ./internal/consensus
```

If you are running this test repeatedly on macOS, you will probably want to do this first, as the KV store is profligate with descriptors. 🦝

```sh
ulimit -n unlimited
```

## Observe

```
--- FAIL: TestReactorValidatorSetChanges (36.80s)
    reactor_test.go:201: 
        	Error Trace:	reactor_test.go:201
        	            				reactor_test.go:210
        	            				asm_amd64.s:1581
        	Error:      	Received unexpected error:
        	            	tx already exists in cache
        	Test:       	TestReactorValidatorSetChanges
FAIL
```

--- a/internal/consensus/reactor_test.go
+++ b/internal/consensus/reactor_test.go
@@ -2,6 +2,7 @@ package consensus
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"os"
 	"path"
@@ -198,7 +199,11 @@ func waitForAndValidateBlock(
 		require.NoError(t, validateBlock(newBlock, activeVals))
 
 		for _, tx := range txs {
-			require.NoError(t, assertMempool(t, states[j].txNotifier).CheckTx(ctx, tx, nil, mempool.TxInfo{}))
+			err := assertMempool(t, states[j].txNotifier).CheckTx(ctx, tx, nil, mempool.TxInfo{})
+			if errors.Is(err, types.ErrTxInCache) {
+				continue
+			}
+			require.NoError(t, err)
 		}
 	}
 

